internal/internal_task_pollers.go

```diff
@@ -564,18 +564,8 @@ func (lath *localActivityTaskHandler) executeLocalActivityTask(task *localActivi
 		return &localActivityResult{task: task, err: err}
 	}
 
-	timeout := task.params.ScheduleToCloseTimeout
-	if task.params.StartToCloseTimeout != 0 && task.params.StartToCloseTimeout < timeout {
-		timeout = task.params.StartToCloseTimeout
-	}
-	timeoutDuration := timeout
-	deadline := time.Now().Add(timeoutDuration)
-	if task.attempt > 1 && !task.expireTime.IsZero() && task.expireTime.Before(deadline) {
-		// this is attempt and expire time is before SCHEDULE_TO_CLOSE timeout
-		deadline = task.expireTime
-	}
```

Comment:
I don't see anything obvious, but are there any ways that you can think of that this logic has changed with your new logic?

Reply:
The only change here is using the scheduled time + `ScheduleToCloseTimeout` when calculating the deadline (used to be `time.Now`). It probably won't effect many activities because for the  first attempt `scheduled time` will be close to `time.Now` and subsequent attempts `task.expireTime` uses workflow time so it will always be shorter than either deadline

Reply:
So if today I have a schedule to close timeout of 5s, my second attempt would get the full 5s (even though that's improper)? But after this change, it will only get however much time since the beginning of the first attempt? If so, I may hit you up off-PR to communicate with one or two known heavy local activity users to confirm this is acceptable.

Reply:
>So if today I have a schedule to close timeout of 5s, my second attempt 

Today you would have no second attempt because  schedule to close was not retried

Reply:
Confirmed off-PR that second-attempt after first-attempt-immediate-failure is bounded today by `expireTime` regardless.